portal/src/hook/usePivot.ts

```diff
@@ -1,24 +1,28 @@
-import { useCallback } from "react";
+import { RefObject, useCallback, createRef } from "react";
 import { useLocation, useNavigate } from "react-router-dom";
-import { IPivotItemProps } from "@fluentui/react";
+import { IPivotItemProps, PivotBase } from "@fluentui/react";
 
 export function usePivotNavigation(): {
   selectedKey: string;
   onLinkClick: (item?: { props: IPivotItemProps }) => void;
+  pivotRef: RefObject<PivotBase>;
 } {
+  const pivotRef = createRef<PivotBase>();
   const navigate = useNavigate();
   const location = useLocation();
   const hash = location.hash.slice(1);
 
   const onLinkClick = useCallback(
-    (item) => {
+    (item?: { props: IPivotItemProps }) => {
       const itemKey = item?.props.itemKey;
       if (typeof itemKey === "string") {
-        navigate(`./#${encodeURIComponent(itemKey)}`);
+        if (itemKey !== pivotRef.current?.state.selectedKey) {
```

Comment:
So the `selectedKey` from the argument is different from `pivotRef.current?.state.selectedKey`?  The `ref.state.selectedKey` seems very hacky to me. It looks like accessing the internal state of the Pivot.

Reply:
> So the selectedKey from the argument is different from pivotRef.current?.state.selectedKey?

Yes, when we click cancel in navigation blocker dialog, `selectedKey` state is updated (unexpected) even if hash (`selectedKey` from the argument) stays the same, and the `PivotItem` is unchanged as well (expected)

> It looks like accessing the internal state of the Pivot.

It is the internal state of Pivot. But I am not sure why the internal state does not match with the blue underline style and the rendered PivotItem